includes/abstracts/abstract-wc-log-handler.php

```diff
@@ -0,0 +1,135 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+if ( ! class_exists( 'WC_Logger' ) ) {
+	include_once( dirname( dirname( __FILE__ ) ) . '/class-wc-logger.php' );
+}
+
+/**
+ * Abstract WC Log Handler Class
+ *
+ * @version        2.8.0
+ * @package        WooCommerce/Abstracts
+ * @category       Abstract Class
+ * @author         WooThemes
+ */
+abstract class WC_Log_Handler {
+
+	/**
+	 * Minimum log level this handler will process.
+	 *
+	 * @var int 0-8 minimum level severity for handling log entry.
+	 * @access private
+	 */
+	protected $_threshold;
+
+	/**
+	 * Log levels by severity.
+	 *
+	 * @var array
+	 * @access private
+	 */
+	protected static $_log_levels = array(
+		WC_Logger::DEBUG     => 0,
+		WC_Logger::INFO      => 1,
+		WC_Logger::NOTICE    => 2,
+		WC_Logger::WARNING   => 3,
+		WC_Logger::ERROR     => 4,
+		WC_Logger::CRITICAL  => 6,
+		WC_Logger::ALERT     => 7,
+		WC_Logger::EMERGENCY => 8,
+	);
+
+	/**
+	 * Constructor for log handler.
+	 *
+	 * @param arr $args {
+	 *     @type string $threshold Optional. Default 'emergency'. Sets the log severity threshold.
+	 *         emergency|alert|critical|error|warning|notice|info|debug
+	 * }
+	 */
+	public function __construct( $args = array() ) {
+
+		$args = wp_parse_args( $args, array(
+			'threshold' => WC_Logger::EMERGENCY,
```

Comment:
@sirreal I understand your point and still is just a abstract class and should override inside the other methods.
Not a problem for me.

But just for you have some idea of what people are using it today:

PayPal example:

```
11-10-2016 @ 14:27:08 - Generating payment form for order 450. Notify URL: http://woo.dev/wc-api/WC_Gateway_Paypal/
11-10-2016 @ 14:27:08 - PayPal Request Args for order 450: cmd=_cart&business=XXXXXXXXXx
```

Stripe example:

```
10-20-2016 @ 21:26:31 - Info: Begin processing payment for order 439 for the amount of 37.70
10-20-2016 @ 21:26:31 - charges request: Array
(
    [currency] => brl
    [amount] => 3770
    [description] => WooCommerce Dev - Order 439
    [capture] => true
    [expand[]] => balance_transaction
    [source] => tok_196pvg4p6qOhgF7yA4iFfJPa
)

10-20-2016 @ 21:26:32 - Processing response: stdClass Object
(
    [id] => ch_196pvi4p6qOhgF7ydnQ3ML8N
    [object] => charge
    [amount] => 3770
    ...
    [status] => succeeded
)

10-20-2016 @ 21:26:49 - Success: Stripe charge complete (Charge ID: ch_196pvi4p6qOhgF7ydnQ3ML8N)
```

So you need to be careful if you display a message like "EMERGENCY" or "CRITICAL" by default in from a debug message, since the users can get really concerned about it.